public/app/features/scenes/core/SceneDataProvider.tsx

```diff
@@ -0,0 +1,78 @@
+import { Unsubscribable } from 'rxjs/internal/types';
+
+import { TimeRange } from '@grafana/data';
+
+import { DataQueryExtended, SceneQueryRunner } from '../querying/SceneQueryRunner';
+
+import { SceneDataObject } from './SceneObjectBase';
+import { SceneDataState } from './types';
+
+export interface SceneDataProviderState extends SceneDataState {
+  queries: DataQueryExtended[];
+}
+
+export class SceneDataProvider extends SceneDataObject<SceneDataProviderState> {
+  private runner: SceneQueryRunner = new SceneQueryRunner({ queries: [] });
```

Comment:
I prefer to have SceneQueryRunner be the name for the SceneObject, makes it much clear what it's doing in the scene object graph. SceneDataProvider feels more like an interface / generic concept, and we already have that in the interface `SceneObject<SceneDataState>` (we could alias a type it like:
```ts
type SceneDataProvider = SceneObject<SceneDataState>
```